internal/mage/util/util.go

```diff
@@ -18,6 +18,9 @@ const (
 func Repository(c *dagger.Client) *dagger.Directory {
 	return c.Host().Directory(".", dagger.HostDirectoryOpts{
 		Exclude: []string{
+			".git",
```

Comment:
If any of the SDK publish workflows rely on getting the git tag/commit/etc. then I think this change will cause them to fail. But I'm not sure if any of them do, so this may be FUD. Just something to double check and be aware of.

Reply:
Yes, I was thinking about that actually, but I wasn't sure. Checking the publishing code now, thanks for the flag!

Reply:
The only publishing part that runs in dagger is the CLI, which **does not** use this `Repository` function:

https://github.com/dagger/dagger/blob/0e8ab1a90fd0fe8ddc6c4e591af164cfe8f3f0aa/internal/mage/cli.go#L26-L40

If you think otherwise @marcosnils, let's re-open this.

I think that this is OK to merge as is, while taking a mental note for the next time that we release.

Reply:
@marcosnils mentioned about `vcsRevision()`:

https://github.com/dagger/dagger/blob/0e8ab1a90fd0fe8ddc6c4e591af164cfe8f3f0aa/internal/engine/version.go#L55-L69

This is only used once, at load time, which means that we are good - it doesn't run in dagger, so the change in `Repository()` should not affect it:

https://github.com/dagger/dagger/blob/0e8ab1a90fd0fe8ddc6c4e591af164cfe8f3f0aa/internal/engine/version.go#L15

Reply:
Awesome! if that's the case, SGTM